### ISSUE TITLE
IndexError: list index out of range

### ISSUE THREAD

**Issue by reviewher** [masaccio/numbers-parser#17]
Rename attachment to .numbers

[numbers_parser_issue.zip](https://github.com/masaccio/numbers-parser/files/8331035/numbers_parser_issue.zip)

```
  File "/usr/local/lib/python3.9/site-packages/numbers_parser/model.py", line 131, in owner_id_map
    ce_id = self.find_refs("CalculationEngineArchive")[0]
IndexError: list index out of range
```

**Comment by masaccio** [comment]
Which version of Numbers was that created with? That object is a fundamental one in my model for parsing the files and if I resave with the latest Numbers the problem goes away. So just trying to understand how I can workaround this.

**Comment by masaccio** [comment]
I use it to detect and manage merged cells and extract functions so I'd hope it's included when those exist. I can easily enough exclude the check for these assuming the calculation_engine reference will appear when merged cells exist or functions exist.

**Comment by reviewher** [comment]
It's probably not required here because the lone table has a single numeric cell without a formula.  In theory there should be no forward references from the cell storage to calculation engine.

That said, it's fair to decide that these files are invalid.  LibreOffice can't open these files either :(

**Comment by masaccio** [comment]
I will add some protection of this and stop the library falling over as that's safe. No need for me to raise an exception

**Comment by masaccio** [comment]
Fixed in 2.3.7
